libs/partners/mongodb/langchain_mongodb/index.py

```diff
@@ -0,0 +1,164 @@
+import logging
+from typing import Any, Dict, List, Optional
+
+from pymongo.collection import Collection
+
+logger = logging.getLogger(__file__)
+
+
+def _create_index_definition(
+    dimensions: int,
+    path: str,
+    similarity: str,
+    filters: Optional[List[Dict[str, str]]],
+) -> Dict[str, Any]:
+    return {
+        "fields": [
+            {
+                "numDimensions": dimensions,
+                "path": path,
+                "similarity": similarity,
+                "type": "vector",
+            },
+            *(filters or []),
+        ]
+    }
+
+
+def ensure_vector_search_index(
+    collection: Collection,
+    index_name: str,
+    dimensions: int,
+    path: str,
+    similarity: str,
+    filters: List[Dict[str, str]],
+    update: bool = False,
+) -> None:
+    """Creates a MongoDB Atlas Vector Search Index
+
+    Args:
+        collection (Collection): MongoDB Collection
+        index_name (str): Name of Index
+        dimensions (int): Number of dimensions in embedding
+        path (str): field with vector embedding
+        similarity (str): The similarity score used for the index
+        filters (List[Dict[str, str]]): additional filters for index definition.
+        update (bool, optional): Update Existing Search index. Defaults to False.
+    """
+    index_exists = bool(list(collection.list_search_indexes(index_name)))
+
+    if update:
+        if index_exists:
+            update_vector_search_index(
+                collection,
+                index_name,
+                dimensions,
+                path,
+                similarity,
+                filters,
+            )
+        else:
+            logger.info(
+                "Index %s does not exist."
+                " will not attempt to update;"
+                " consider setting update = False to create"
+                " a new vector search index",
+                index_name,
+            )
+
+    else:
+        if not index_exists:
+            create_vector_search_index(
+                collection,
+                index_name,
+                dimensions,
+                path,
+                similarity,
+                filters,
+            )
+        else:
+            logger.info(
+                "Index %s already exists,"
```

Comment:
I would expect this to raise an error.

Reply:
Removing `index_exists`. Placing the error message here for posterity:
```
pymongo.errors.OperationFailure: Duplicate Index, full error: {'ok': 0.0, 'errmsg': 'Duplicate Index', 'code': 68, 'codeName': 'IndexAlreadyExists', '$clusterTime': {'clusterTime': Timestamp(1710988691, 1), 'signature': {'hash': b'=\xa5\xf1\xb3\x8c\xc3`\xaf\xe8"\x8a\x8c7\xac\xb2\xd9\t\xd9\xf8\xb9', 'keyId': 7292037883655880706}}, 'operationTime': Timestamp(1710988691, 1)}
```